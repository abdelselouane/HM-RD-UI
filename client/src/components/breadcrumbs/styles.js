import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
