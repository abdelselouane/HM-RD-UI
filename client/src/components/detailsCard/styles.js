import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
