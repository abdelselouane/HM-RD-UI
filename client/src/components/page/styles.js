import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
