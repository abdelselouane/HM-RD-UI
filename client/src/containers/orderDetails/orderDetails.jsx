import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { withContext } from '../../contexts/history';

import Page from '../../components/page';
import DetailsCard from '../../components/detailsCard';

import { getFormattedDate } from '../../utils/formatHelper';

import { Container, CardTitle, PurchaseOrderList } from './styles';


export class OrderDetails extends Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        partsOrderId: PropTypes.string.isRequired
      }).isRequired
    }).isRequired,
    orders: PropTypes.arrayOf(
      PropTypes.shape({
        partsOrderId: PropTypes.number.isRequired
      })
    ).isRequired
  };

  state = {
    order: null
  };

  componentWillMount() {
    const { partsOrderId } = this.props.match.params;
    const index = this.props.orders.findIndex(
      order => order.partsOrderId === Number.parseInt(partsOrderId, 10)
    );
    this.setState({ order: index > -1 ? this.props.orders[index] : null });
  }

  getDetailsCardsProps = partsDetails => {
    const mappedPoNbrs = [];
    const detailsCards = [];
    for (let i = 0; i < partsDetails.length; i += 1) {
      if (!mappedPoNbrs.includes(partsDetails[i].purchaseOrderNbr)) {
        const cardProps = {
          purchaseOrderNbr: partsDetails[i].purchaseOrderNbr,
          purchaseOrderStatDesc: partsDetails[i].purchaseOrderStatDesc,
          parts: []
        };
        for (let j = i; j < partsDetails.length; j += 1) {
          const { purchaseOrderNbr, purchaseOrderStatDesc, ...partInfo } = partsDetails[j];
          if (partsDetails[i].purchaseOrderNbr === purchaseOrderNbr) {
            cardProps.parts.push(partInfo);
          }
        }
        mappedPoNbrs.push(partsDetails[i].purchaseOrderNbr);
        detailsCards.push(cardProps);
      }
    }

    return detailsCards;
  }

  renderDetailsCards = order => {
    const { partsDetails } = order;
    if (!partsDetails) {
      return <div>Purchase Order details could not be loaded</div>;
    }
    return this.getDetailsCardsProps(partsDetails).map(props =>
      <DetailsCard key={props.purchaseOrderNbr} {...props} />
    );
  };

  render() {
    const { order } = this.state;
    return (
      <Page className="order-details" showReturn>
        <Container>
          {order ? (
            <React.Fragment>
              <CardTitle className="title">
                <h2>{`Order #${order.partsOrderId || 0} Details`}</h2>
                <span>{`Submitted ${getFormattedDate(order.orderCreatedDate)}`}</span>
              </CardTitle>
              <PurchaseOrderList>{this.renderDetailsCards(order)}</PurchaseOrderList>
            </React.Fragment>
          ) : (
            <div className="no-results">Order Request not found</div>
          )}
        </Container>
      </Page>
    );
  }
}

const mapStateToProps = state => ({
  orders: state.partsOrders.orders
});

const OrderDetailsWithHistory = withContext(OrderDetails);

export default connect(
  mapStateToProps,
  null
)(OrderDetailsWithHistory);
