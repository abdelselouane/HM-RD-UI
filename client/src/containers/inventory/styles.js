import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
