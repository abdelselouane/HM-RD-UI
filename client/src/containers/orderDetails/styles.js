import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
