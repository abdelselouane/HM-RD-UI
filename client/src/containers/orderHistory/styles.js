import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;

  .title {
    padding: 0px 20px;
  }
  
  .order-history-header {
    margin: 0px;
  
    thead tr {
      margin-left: 1em;
  
      th {
        border: none;
        font-size: 18px;
  
        &:first-child {
          padding-left: 20px;
        }
      }
    }
  }
`;

export default Container;
