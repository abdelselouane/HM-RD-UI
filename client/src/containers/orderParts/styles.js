import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background: ${({ theme }) => theme.white};
`;

export default Container;
