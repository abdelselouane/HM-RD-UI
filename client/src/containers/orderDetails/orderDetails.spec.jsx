/* global describe, beforeEach, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import * as Styles from './styles';

import { OrderDetails } from './orderDetails';

describe('<OrderDetails />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      match: {
        params: {
          partsOrderId: '2'
        }
      },
      orders: [
        {
          orderCreatedDate: '1970-12-02',
          partsOrderId: 2,
          workOrderNbr: '',
          partsDetails: [
            {
              purchaseOrderNbr: '971070001',
              purchaseOrderStatDesc: 'SUBMITTED',
              partDescription: 'mockDescription3',
              partNbr: 'abc-123',
              brandName: 'mockbrand',
              quantity: 1
            },
            {
              purchaseOrderNbr: '971070001',
              purchaseOrderStatDesc: 'SUBMITTED',
              partDescription: 'mockDescription4',
              partNbr: 'cba-321',
              brandName: 'mockbrand',
              quantity: 2
            },
            {
              purchaseOrderNbr: '971070002',
              purchaseOrderStatDesc: 'SUBMITTED',
              partDescription: 'mockDescription1',
              partNbr: '01-23456',
              brandName: 'brandy',
              quantity: 4
            },
            {
              purchaseOrderNbr: '971070002',
              purchaseOrderStatDesc: 'SUBMITTED',
              partDescription: 'mockDescription2',
              partNbr: '01-09897',
              brandName: 'brandy',
              quantity: 2
            }
          ]
        },
        {
          orderCreatedDate: '1970-12-01',
          partsOrderId: 1,
          workOrderNbr: '123456789',
          partsDetails: [
            {
              purchaseOrderNbr: '971070002',
              purchaseOrderStatDesc: 'SUBMITTED',
              partDescription: 'mockDescription1',
              partNbr: '01-23456',
              brandName: 'brandy',
              quantity: 4
            },
            {
              purchaseOrderNbr: '971070002',
              purchaseOrderStatDesc: 'SUBMITTED',
              partDescription: 'mockDescription2',
              partNbr: '01-09897',
              brandName: 'brandy',
              quantity: 2
            }
          ]
        }
      ]
    };

    wrapper = shallow(<OrderDetails {...props} />);
  });

  it('should render properly for a single Purchase Order', () => {
    wrapper.setProps({ match: { params: { partsOrderId: '1' } } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly for 2 Purchase Orders', () => {
    wrapper.setProps({ match: { params: { partsOrderId: '2' } } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly when partsDetails is not defined', () => {
    wrapper.setState({ order: { ...props.orders[0], partsDetails: [] } });
    // console.log(wrapper.state().order);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly if partsOrderId is not in the order list', () => {
    wrapper.setProps({ match: { params: { partsOrderId: '8' } } });
    const instance = wrapper.instance();
    instance.componentWillMount();
    expect(instance.state.order).toBe(null);
  });

  it('should render properly if order is null', () => {
    wrapper.setState({ order: null });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });
  });
});
