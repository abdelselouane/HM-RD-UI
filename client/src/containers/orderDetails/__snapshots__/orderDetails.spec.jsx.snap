// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetails /> should render properly for 2 Purchase Orders 1`] = `
<Connect(withContext(Page))
  className="order-details"
  showReturn={true}
>
  <styled.div>
    <styled.div
      className="title"
    >
      <h2>
        Order #2 Details
      </h2>
      <span>
        Submitted Dec 02, 1970
      </span>
    </styled.div>
    <styled.div>
      <DetailsCard
        key="971070001"
        parts={
          Array [
            Object {
              "brandName": "mockbrand",
              "partDescription": "mockDescription3",
              "partNbr": "abc-123",
              "quantity": 1,
            },
            Object {
              "brandName": "mockbrand",
              "partDescription": "mockDescription4",
              "partNbr": "cba-321",
              "quantity": 2,
            },
          ]
        }
        purchaseOrderNbr="971070001"
        purchaseOrderStatDesc="SUBMITTED"
      />
      <DetailsCard
        key="971070002"
        parts={
          Array [
            Object {
              "brandName": "brandy",
              "partDescription": "mockDescription1",
              "partNbr": "01-23456",
              "quantity": 4,
            },
            Object {
              "brandName": "brandy",
              "partDescription": "mockDescription2",
              "partNbr": "01-09897",
              "quantity": 2,
            },
          ]
        }
        purchaseOrderNbr="971070002"
        purchaseOrderStatDesc="SUBMITTED"
      />
    </styled.div>
  </styled.div>
</Connect(withContext(Page))>
`;

exports[`<OrderDetails /> should render properly for a single Purchase Order 1`] = `
<Connect(withContext(Page))
  className="order-details"
  showReturn={true}
>
  <styled.div>
    <styled.div
      className="title"
    >
      <h2>
        Order #2 Details
      </h2>
      <span>
        Submitted Dec 02, 1970
      </span>
    </styled.div>
    <styled.div>
      <DetailsCard
        key="971070001"
        parts={
          Array [
            Object {
              "brandName": "mockbrand",
              "partDescription": "mockDescription3",
              "partNbr": "abc-123",
              "quantity": 1,
            },
            Object {
              "brandName": "mockbrand",
              "partDescription": "mockDescription4",
              "partNbr": "cba-321",
              "quantity": 2,
            },
          ]
        }
        purchaseOrderNbr="971070001"
        purchaseOrderStatDesc="SUBMITTED"
      />
      <DetailsCard
        key="971070002"
        parts={
          Array [
            Object {
              "brandName": "brandy",
              "partDescription": "mockDescription1",
              "partNbr": "01-23456",
              "quantity": 4,
            },
            Object {
              "brandName": "brandy",
              "partDescription": "mockDescription2",
              "partNbr": "01-09897",
              "quantity": 2,
            },
          ]
        }
        purchaseOrderNbr="971070002"
        purchaseOrderStatDesc="SUBMITTED"
      />
    </styled.div>
  </styled.div>
</Connect(withContext(Page))>
`;

exports[`<OrderDetails /> should render properly if order is null 1`] = `
<Connect(withContext(Page))
  className="order-details"
  showReturn={true}
>
  <styled.div>
    <div
      className="no-results"
    >
      Order Request not found
    </div>
  </styled.div>
</Connect(withContext(Page))>
`;

exports[`<OrderDetails /> should render properly when partsDetails is not defined 1`] = `
<Connect(withContext(Page))
  className="order-details"
  showReturn={true}
>
  <styled.div>
    <styled.div
      className="title"
    >
      <h2>
        Order #2 Details
      </h2>
      <span>
        Submitted Dec 02, 1970
      </span>
    </styled.div>
    <styled.div />
  </styled.div>
</Connect(withContext(Page))>
`;

exports[`<OrderDetails /> should render styles properly 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`<OrderDetails /> should render styles properly 2`] = `
.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;
}

.c0 h2 {
  display: inline-block;
  margin-bottom: 0px;
}

<div
  className="c0"
/>
`;

exports[`<OrderDetails /> should render styles properly 3`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 20px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
}

<div
  className="c0"
/>
`;
